mbofui/app/app.js

```diff
@@ -0,0 +1,224 @@
+          name: 'OpenStreetMap',
+          url: 'http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+          type: 'xyz'
+        },
+        terrain: {
+          name: 'Terrain',
+          url: 'http://stamen-tiles-{s}.a.ssl.fastly.net/terrain-background/{z}/{x}/{y}.png',
+          type: 'xyz',
+        },
+        osmh: {
+          name: 'OpenStreetMap Hot',
+          type: 'xyz',
+          url: 'http://{s}.tile.openstreetmap.fr/hot/{z}/{x}/{y}.png',
+        },
+        esriwsm: {
+          name: 'Esri World StreetMap',
+          type: 'xyz',
+          url: 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}'
+        },
+        esrisatellite: {
+          name: 'Esri Satellite',
+          type: 'xyz',
+          url: 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}'
+        }
+      },
+      overlays: {
+        bofs: {
+          name: "Bofs",
+          type: "markercluster",
+          visible: true
+        },
+        draw: {
+          name: 'draw',
+          type: 'group',
+          visible: true,
+          layerParams: {
+            showOnSelector: false
+          }
+        }
+      },
+      awesomeMarkerIcon: {
+        type: 'awesomeMarker',
+        markerColor: 'blue'
+      },
+    }
+  } );
+
+  $scope.categories = null;
+
+  $scope.bofIt = function () {
+    leafletData.getMap().then( function ( map ) {
+      map.locate( {
+        setView: true,
+        maxZoom: 16
+      } );
+
+      function onLocationFound( e ) {
+        var radius = e.accuracy / 2;
+        $( '#bofModal' ).attr( 'data-coords', [ e.latlng.lat, e.latlng.lng ] );
+        L.popup()
+          .setLatLng( e.latlng )
+          .setContent( '<a href="" data-toggle="modal" data-target="#bofModal">Add an event here?</a>' )
+          .openOn( map );
+      }
+      map.on( 'locationfound', onLocationFound );
+    } );
+  };
+
+  $scope.markers = [];
+  $scope.markersAll = [];
+
+  bofDataRef.on( 'child_added', function ( snapshot ) {
+    var marker = snapshot.val();
+    var key = snapshot.key();
+    newMarker = {
+      fbid: key,
+      lat: parseFloat( marker.lat ),
+      lng: parseFloat( marker.long ),
+      category: marker.category,
+      message: resolveCategory( marker.category ) + '<div>' + marker.what + '</div>',
+      layer: 'bofs',
+      icon: resolveIcon( marker.category ),
+      draggable: marker.draggable
+    };
+    $scope.markersAll.push( newMarker );
+    $scope.markers.push( newMarker );
+  } );
+
+
+  $scope.$on( "leafletDirectiveMap.contextmenu", function ( event, args ) {
+    leafletData.getMap().then( function ( map ) {
+      var leafEvent = args.leafletEvent;
+      $( '#bofModal' ).attr( 'data-coords', [ leafEvent.latlng.lat, leafEvent.latlng.lng ] );
+      L.popup()
+        .setLatLng( leafEvent.latlng )
+        .setContent( '<a href="" data-toggle="modal" data-target="#bofModal">Add an event here?</a>' )
```

Comment:
I often feel bad about HTML tags appearing in JS code (and other languages).  Since you're already using jQuery, why not use `$()` to create elements along with its `attr()` and other methods to set the attributes, content, etc.?